src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerWrapper.java

```diff
@@ -0,0 +1,84 @@
+import java.util.concurrent.CountDownLatch;
+
+import static uk.gov.companieshouse.itemhandler.ItemHandlerApplication.APPLICATION_NAMESPACE;
+
+@Aspect
+@Service
+public class OrdersKafkaConsumerWrapper {
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
+    private CountDownLatch latch = new CountDownLatch(1);
+    private String orderUri;
+    private String testType = "DEFAULT";
+    @Value("${kafka.broker.addresses}")
+    private String brokerAddresses;
+    private KafkaTemplate<String, String> template;
+    private static final String ORDER_RECEIVED_TOPIC = "order-received";
+    private static final String ORDER_RECEIVED_TOPIC_RETRY = "order-received-retry";
+    private static final String ORDER_RECEIVED_TOPIC_ERROR = "order-received-error";
+    private static final String ORDER_RECEIVED_URI = "/order/ORDER-12345";
+
+    @Before(value = "execution(* uk.gov.companieshouse.itemhandler.kafka.OrdersKafkaConsumer.processOrderReceived(..)) && args(message)")
+    public void beforeOrderProcessed(final String message) throws Exception {
+        LOGGER.info("OrdersKafkaConsumer.processOrderReceived() @Before triggered");
+        // mock message processing failure scenario for main listener
+        // so that the message can be published to alternate topics '-retry' and '-error'
+        if (this.testType.equals("RETRY")) {
+            throw new Exception("Mock main listener exception");
+        }
+    }
+
+    @AfterThrowing(pointcut = "execution(* uk.gov.companieshouse.itemhandler.kafka.OrdersKafkaConsumer.*(..))", throwing = "x")
+    public void orderProcessedException(final Exception x) throws Throwable {
+        LOGGER.info("OrdersKafkaConsumer.processOrderReceived() @AfterThrowing triggered");
+
+        // mock exception handler to demonstrate publishing of unprocessed
+        // message to alternate topics '-retry' and '-error'
+        setUpTestKafkaOrdersProducerAndSendMessageToTopic();
+    }
+
+    @After(value = "execution(* uk.gov.companieshouse.itemhandler.kafka.OrdersKafkaConsumer.*(..)) && args(message)")
+    public void afterOrderProcessed(final String message){
+        LOGGER.info("OrdersKafkaConsumer.processOrderReceivedRetry() @After triggered");
+        latch.countDown();
+        this.orderUri = message;
+    }
+
+    CountDownLatch getLatch() { return latch; }
+    String getOrderUri() { return orderUri; }
+    void setTestType(String type) { this.testType = type;}
+
+    private void setUpTestKafkaOrdersProducerAndSendMessageToTopic() {
+        final Map<String, Object> senderProperties = KafkaTestUtils.senderProps(brokerAddresses);
+
+        senderProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+        final ProducerFactory<String, String> producerFactory = new DefaultKafkaProducerFactory<>(senderProperties);
+
+        template = new KafkaTemplate<>(producerFactory);
+        if (this.testType.equals("RETRY")) {
```

Comment:
The way `testType` is being used suggests it might be better represented by an `Enum`? Possibly `CHConsumerType` even?